### PR TITLE
Improved Stable Diffusion optimization example

### PR DESCRIPTION
**Pull request by AlexKoff88**
# What does this PR do?
- Added support for more datasets
- Accelerated load of laion* datasets
- Updated README and notebook

### CODE REVIEW

File: examples/openvino/stable-diffusion/README.md [hunk 1]
```diff
@@ -1,15 +1,17 @@
...
+This example supports model tuning on the following datasets from the HuggingFace:
 * [Pokemon BLIP captions](https://huggingface.co/datasets/lambdalabs/pokemon-blip-captions)
 * [laion2B-en](https://huggingface.co/datasets/laion/laion2B-en)
 * [laion2B-en-aesthetic](https://huggingface.co/datasets/laion/laion2B-en-aesthetic)
+* [laion-art](https://huggingface.co/datasets/laion/laion-art)
+* [laion400m](https://huggingface.co/datasets/laion/laion400m)
 
 But it can be easily extended to other datasets.
->**Note**: laion2B-en is being downloaded on-fly durint the fine-tuning process. No need to store it locally.
+>**Note**: laion2B* datasets are being downloaded on-fly durint the fine-tuning process. No need to store them locally.
```
Comment (echarlaix) [inline examples/openvino/stable-diffusion/README.md:14]:
```suggestion
>**Note**: laion2B* datasets are being downloaded on-fly during the fine-tuning process. No need to store them locally.
```

File: examples/openvino/stable-diffusion/train_text_to_image_qat.py [hunk 2]
```diff
@@ -882,6 +971,7 @@ def collate_fn(examples):
     # Reinit
     optimizer = optimizer_cls(
         filter(lambda p: p.requires_grad, unet.parameters()),
+        # gain_or_bias_params,
```
Comment (echarlaix) [inline examples/openvino/stable-diffusion/train_text_to_image_qat.py:974]:
```suggestion
```

File: examples/openvino/stable-diffusion/train_text_to_image_qat.py [hunk 3]
```diff
@@ -397,6 +422,13 @@ def parse_args():
         choices=["cpu", "cuda"],
         help="Whether to use EMA model and where to store the EMA model.",
     )
+    parser.add_argument(
+        "--quantization_mode",
+        type=str,
+        default="moderate",
+        choices=["moderate", "aggressive"],
+        help="Whether to use EMA model and where to store the EMA model.",
```
Comment (echarlaix) [inline examples/openvino/stable-diffusion/train_text_to_image_qat.py:430]: Could you add some information on what component will be impacted (attention,...)

**Review by echarlaix — APPROVED** [review]
LGTM !

File: examples/openvino/stable-diffusion/train_text_to_image_qat.py (hunk 3 above)
Comment (AlexKoff88) [reply examples/openvino/stable-diffusion/train_text_to_image_qat.py]: Thank you. Updated.

**Comment by echarlaix** [comment]
Will merge #326 as failing tests are not related to modifications

**Pull request merged**